MavenArtifacts.md

```diff
@@ -0,0 +1,83 @@
+
+## Artifact Names
+
+WPILib builds 4 different types of artifacts.
+
+##### C++ Only Libraries
+When we publish C++ only libraries, they are published with the just the base artifact name as their artifact name. All dependencies for the library are shared linked into the binary.
+
+#### Java Only Libraries
+When we publish Java only libraries, they are published with the just the base artifact name as their artifact name.
+
+#### C++/Java Libraries without JNI
+For libraries that are both C++ and Java, but without a JNI component, the C++ component is published with the `basename-cpp` artifact name, and the Java component is published with the `basename-java` artifact name.
+
+#### C++/Java Libraries with JNI
+For libraries that are both C++ and Java with a JNI component there are 3 different artifact names. For Java, the component is published as `basename-java`. For C++, the `basename-cpp` artifact contains the C++ artifacts with all dependencies shared linked into the binary. These binaries DO contain the JNI entry points. The `basename-jni` artifact contains identical C++ binaries to the `-cpp` artifact, however all of its dependencies are statically linked, and only the JNI and C entry points are exported.
+
+The `-jni` artifact should only be used in cases where you want to create a self contained Java application where the native artifacts are embedded in the jar. Note in an extraction scenario, extending off of the library is never supported, which is why the C++ entry points are not exposed. The name of the library is randomly generated during extraction. For pretty much all cases, and if you ever want to extend from a native library, you should use the `-cpp` artifacts. GradleRIO uses the `-cpp` artifacts for all platforms, even desktop, for this reason.
+
+ ## Provided Artifacts
+This repository provides the following artifacts. Below each artifact is its dependencies. Note if ever using the `-jni` artifacts, no dependencies are needed for native binaries.
+
+For C++, if building with static dependencies, the listed order should be the link order in your linker.
+
+All artifacts are based at `edu.wpi.first.artifactname` in the repository.
+
+wpiutil
```

Comment:
For these, do you want to use a level 4 heading?

Reply:
Would probably look better with top-level bullets and sub-bullets though, as these are actually a list. It would probably work even better as a table with the top-level on the left and subartifacts on the right, but markdown tables are terrible to make so I won't ask that of you :).